.github/workflows/release.yaml

```diff
@@ -0,0 +1,105 @@
+name: CI on Releasing Tag
+
+on:
+  push:
+    tags:
+      - "*"
+
+env:
+  IMAGE: rancher/system-agent-installer-k3s
+
+jobs:
+  build-linux:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write # needed for the Vault authentication
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ linux ]
+        arch: [ amd64, arm64 ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Fix the not-a-git-repository issue
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+
+      - name: Set environment variables
+        run: |
+          echo "ARCH=${{ matrix.arch }}" >> "$GITHUB_ENV"
+          echo "OS=${{ matrix.os }}"     >> "$GITHUB_ENV"
+
+      - name: Download installer
+        run: scripts/download
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Load secrets from Vault
+        uses: rancher-eio/read-vault-secrets@main
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ env.DOCKER_USERNAME }}
+          password: ${{ env.DOCKER_PASSWORD }}
+
+      - name: Build and push Docker image
+        id: build
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: package/Dockerfile
+          push: true
+          tags: "${{ env.IMAGE }}:${{ env.VERSION }}-${{ env.OS }}-${{ env.ARCH }}"
+          platforms: "${{ env.OS }}/${{ env.ARCH }}"
+
+  create-docker-manifest:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write # needed for the Vault authentication
+    needs: [ build-linux ]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set environment variables
+        run: |
+          . scripts/version
+          echo "VERSION=$VERSION"         >> "$GITHUB_ENV" 
```

Comment:
this is in the scripts/version file too, I think one could be removed?

Reply:
removed